### PR TITLE
use turbo mode for lookup plugins

### PR DESCRIPTION
**Pull request by abikouo**
##### SUMMARY

use turbo mode for lookup plugins

##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by abikouo** [comment]
recheck

**Review by Akasurde — APPROVED** [review]

File: plugins/lookup/cluster_moid.py [hunk 1]
```diff
@@ -49,22 +49,20 @@
...
 class LookupModule(LookupBase):
-    def run(self, terms, variables, **kwargs):
-        import asyncio
-
+    async def _run(self, terms, variables, **kwargs):
         self.set_options(var_options=variables, direct=get_credentials(**kwargs))
         self.set_option("object_type", "cluster")
-        loop = asyncio.get_event_loop()
+        result = await Lookup.entry_point(terms, self._options)
+        return [result]
```
[the remark this answers is not in the document]
Comment (alinabuzachis) [reply plugins/lookup/cluster_moid.py:66]: @abikouo you are still returning a list.

**Review by alinabuzachis — CHANGES_REQUESTED** [review]

File: plugins/lookup/datacenter_moid.py [hunk 2]
```diff
@@ -49,22 +49,20 @@
...
 
 class LookupModule(LookupBase):
-    def run(self, terms, variables, **kwargs):
-        import asyncio
-
+    async def _run(self, terms, variables, **kwargs):
         self.set_options(var_options=variables, direct=get_credentials(**kwargs))
         self.set_option("object_type", "datacenter")
-        loop = asyncio.get_event_loop()
+        result = await Lookup.entry_point(terms, self._options)
```
Comment (alinabuzachis) [inline plugins/lookup/datacenter_moid.py:65]: Even here.

**Review by alinabuzachis — CHANGES_REQUESTED** [review]

**Comment by alinabuzachis** [comment]
@abikouo Thank you for working on this. The result is stil returned as a list.

**Comment by abikouo** [comment]
> @abikouo Thank you for working on this. The result is stil returned as a list and it must be a string.

It is failing when I do not return a list, it seems that the result should be a list, otherwise when the option ``wantList`` is set to true the raw result is returned, otherwise the core engine will perform a join operation.

**Comment by alinabuzachis** [comment]
> > @abikouo Thank you for working on this. The result is stil returned as a list and it must be a string.
> 
> It is failing when I do not return a list, it seems that the result should be a list, otherwise when the option `wantList` is set to true the raw result is returned, otherwise the core engine will perform a join operation.

We agreed to keep it as simple as possible https://github.com/ansible-collections/vmware.vmware_rest/pull/209#issuecomment-878365007 and have the result returned as a string, it doesn't make sense to have the result as a list because only one path is processed at time. If we want to get the moid from multiple paths we use a different approach (as @goneri explained).

**Comment by abikouo** [comment]
> > > @abikouo Thank you for working on this. The result is stil returned as a list and it must be a string.
> > 
> > 
> > It is failing when I do not return a list, it seems that the result should be a list, otherwise when the option `wantList` is set to true the raw result is returned, otherwise the core engine will perform a join operation.
> 
> We agreed to keep it as simple as possible [#209 (comment)](https://github.com/ansible-collections/vmware.vmware_rest/pull/209#issuecomment-878365007) and have the result returned as a string. It doesn't make sense to have the result as a list because only one path is processed at time. If we want to get the moid from multiple paths we use a different approach (as @goneri explained). For this reason the result has been returned and documented as a string.

@alinabuzachis I got the point, but the result should be return as list, this is expected on ansible core otherwise when you set the option ``wantLIst`` to ``true`` it will not work, I think the confusion with the current version if that ``lookup.entry_point`` will return a string result, but when calling ``asyncio.gather`` this is wrapping the result into a list, do no hesitate if you have others questions

**Comment by alinabuzachis** [comment]
> > > > @abikouo Thank you for working on this. The result is stil returned as a list and it must be a string.
> > > 
> > > 
> > > It is failing when I do not return a list, it seems that the result should be a list, otherwise when the option `wantList` is set to true the raw result is returned, otherwise the core engine will perform a join operation.
> > 
> > 
> > We agreed to keep it as simple as possible [#209 (comment)](https://github.com/ansible-collections/vmware.vmware_rest/pull/209#issuecomment-878365007) and have the result returned as a string. It doesn't make sense to have the result as a list because only one path is processed at time. If we want to get the moid from multiple paths we use a different approach (as @goneri explained). For this reason the result has been returned and documented as a string.
> 
> @alinabuzachis I got the point, but the result should be return as list, this is expected on ansible core otherwise when you set the option `wantLIst` to `true` it will not work, I think the confusion with the current version if that `lookup.entry_point` will return a string result, but when calling `asyncio.gather` this is wrapping the result into a list, do no hesitate if you have others questions

`asyncio.gather` was initially used when I supported a list of paths and then returned a list of MoIDs, but I then forgot to remove it. Even if using `wantLIst=true` won't make sense in our case because we have only one result, it makes sense your approach.

**Review by alinabuzachis — APPROVED** [review]

**Pull request merged**